### DISCUSSION TITLE
Where OmittedTypeArgumentSyntax is implemented

### DISCUSSION THREAD

**Discussion by Youssef1313** [dotnet/roslyn#47265] [General] [answered]
I can't find anywhere the implementation of this class in the repo. However, it's documented [here](https://docs.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.csharp.syntax.omittedtypeargumentsyntax?view=roslyn-dotnet).

I was expecting to find a file called *OmittedTypeArgumentSyntax.cs* [here](https://github.com/dotnet/roslyn/tree/4d24556df984774dc70cf707eff1ee9d3d96b9b9/src/Compilers/CSharp/Portable/Syntax).

**Comment by svick** [comment, marked as answer]
If you look at the the directory you linked to, you'll notice that it contains `partial` classes, usually with only a small amount of code, e.g. [here](https://github.com/dotnet/roslyn/blob/4d24556df984774dc70cf707eff1ee9d3d96b9b9/src/Compilers/CSharp/Portable/Syntax/BlockSyntax.cs). The rest of the code is generated from [Syntax.xml](https://github.com/dotnet/roslyn/blob/4d24556df984774dc70cf707eff1ee9d3d96b9b9/src/Compilers/CSharp/Portable/Syntax/Syntax.xml).

The easiest way to look at the generated code is [using sourceroslyn.io](http://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp/Generated/Syntax.xml.Syntax.Generated.cs,0cbaefb698db4d39).
